### PR TITLE
if query is not updated, leave it untouched

### LINKED ISSUE #281: Does 'https://curl.se/?&&&'  really have a blank query?
We currently have a test verifying this, but I disagree. There are clearly three letters in the query and since there is no standard format for queries, a user should be able to extract them even if they are three ampersands.

### PR DESCRIPTION
**Pull request by bagder**
Updated the tests accordingly.

Fixes #281

### CODE REVIEW

File: trurl.c [hunk 1]
```diff
@@ -1215,21 +1217,23 @@ static void extractqpairs(CURLU *uh, struct option *o)
...
+      char *oldnq = nq;
+      nq = curl_maprintf("%s%s%s", nq?nq:"",
+                         (nq && *nq && *(qpairs[i].str))? o->qsep: "",
+                         qpairs[i].str);
+      curl_free(oldnq);
+    }
+    if(nq) {
+      int rc = curl_url_set(uh, CURLUPART_QUERY, nq, 0);
+      if(rc)
+        trurl_warnf(o, "internal problem");
```
Comment (jacobmealey) [inline trurl.c:1233]: i know this isn't a new warning but im wondering if we can make this warning more descriptive?

Comment (jacobmealey) [reply trurl.c:1233]: oops looks like i was a minute late to the party haha

Comment (bagder) [reply trurl.c:1233]: well, we can/should still improve the warning!

Comment (bagder) [reply trurl.c:1233]: it should be extremely rare to see this...

Comment (bagder) [reply trurl.c:1233]: see #285

**Pull request merged**